portfolio/src/main/webapp/js/custom-script.js

```diff
@@ -423,3 +425,57 @@ function drawInteractiveChart() {
         interactiveChart.draw(emotionData, options);
       });
 }
+
+const typedText = document.querySelector('.typed-text');
+/**
+ * TODO Stretch Feature(briafassler): read names from a database instead of an
+ * array
+ */
+const martyrNames = [
+  'George Floyd',          'Breonna Taylor',  'Ahmaud Arbery',
+  'Rayshard Brooks',       'Robert Fuller',   'James Scurlock',
+  'Elijah McClain',        'Justin Howell',   'Jamel Floyd',
+  'Eric Garner',           'Michael Brown',   'Eric Reason',
+  'Sandra Bland',          'Jamar Clark',     'Tamir Rice',
+  'John Crawford III',     'Phillip White',   'Ezell Ford',
+  'Dante Parker',          'Laquan McDonald', 'Dominique Clayton',
+  'Michelle Cusseaux',     'George Mann',     'Tanisha Anderson',
+  'Akai Gurley',           'Alonzo Smith',    'Stephon Clark',
+  'Philando Castile',      'Janet Wilson',    'Darrius Stewart',
+  'Botham Jean',           'Samuel Dubose',   'Willie Tillman',
+  'Quintonio Legrier',     'Brian Keith Day', 'Christian Taylor',
+  'Junior Prosper',        'Freddie Gray',    'Christopher McCorvey',
+  'Peter Gaines',          'Mya Hall',        'Felix Kumi',
+  'Keith Harrison McLeod', 'Anthony Ashford', 'Salvado Ellswood',
+];
+let marytrNamesIndex = 0;
+let charIndex = 0;
+
+function typeName() {
+  if (charIndex < martyrNames[marytrNamesIndex].length) {
+    typedText.textContent += martyrNames[marytrNamesIndex].charAt(charIndex);
+    charIndex++;
+    setTimeout(typeName, 150);
+  } else {
+    setTimeout(eraseName, 2000);
+  }
+}
+
+function eraseName() {
+  if (charIndex > 0) {
+    typedText.textContent =
+        martyrNames[marytrNamesIndex].substring(0, charIndex - 1);
+    charIndex--;
+    setTimeout(eraseName, 100);
+  } else {
+    marytrNamesIndex++;
+    if (marytrNamesIndex >= martyrNames.length) {
+      marytrNamesIndex = 0;
+    }
+    setTimeout(typeName, 1300);
```

Comment:
I really like the recursive aspect of the functions. It's really smart.

Reply:
I agree it's very clever! Wild stuff.

Could you have used setInterval() instead to simplify this? To handle erase vs. delete you'd just need to set a "direction" variable somewhere.

Reply:
Using setInterval() will continue to trigger the expression unless told to stop, so my thinking was that it would continue to print out out names without erasing which was another problem I had when first testing the feature. Using setTimeout allows for the name to print once which is essentially the goal. Any thoughts on this?

Reply:
Hm good point, let's just leave it as is.